### PR TITLE
WFLY-14317 Replace 'Java Message Service|JMS' references in java doc …

### PR DESCRIPTION
**Pull request by RanabirChakraborty**
Issue: https://issues.redhat.com/browse/WFLY-14317
(Replace "Java Message Service|JMS" references in java doc with corresponding Jakarta EE ones)
Link with: https://github.com/wildfly/wildfly/pull/13808

### CODE REVIEW

**Comment by RanabirChakraborty** [comment]
@ehsavoie If you can review this PR :)

CC: @asoldano @scottmarlow

**Review by ehsavoie — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_Backward_&_Forward_Compatibility.adoc [hunk 1]
```diff
@@ -7,21 +7,21 @@ as JBoss AS7, WildFly {wildflyVersion} & 9). +
 These two compatibility modes are provided by the ActiveMQ Artemis
 project that supports the HornetQ's CORE protocol:
 
-* backward compatibility: WildFly 10 JMS clients (using Artemis) can
+* backward compatibility: WildFly 10 Jakarta Messaging clients (using Artemis) can
 connect to legacy app server (running HornetQ)
-* forward compatibility: legacy JMS clients (using HornetQ) can connect
+* forward compatibility: legacy Jakarta Messaging clients (using HornetQ) can connect
```
[the remark this answers is not in the document]
Comment (ehsavoie) [reply docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_Backward_&_Forward_Compatibility.adoc:12]: or just use messaging

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_Backward_&_Forward_Compatibility.adoc [hunk 2]
```diff
@@ -7,21 +7,21 @@ as JBoss AS7, WildFly {wildflyVersion} & 9). +
...
 to WildFly 10 app server (running Artemis).
 
 [[forward-compatibility]]
 == Forward Compatibility
 
-Forward compatibility requires no code change in legacy JMS clients. It
+Forward compatibility requires no code change in legacy Jakarta Messaging clients. It
```
[the remark this answers is not in the document]
Comment (ehsavoie) [reply docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_Backward_&_Forward_Compatibility.adoc:18]: or just use messaging

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_Backward_&_Forward_Compatibility.adoc [hunk 3]
```diff
@@ -49,14 +49,14 @@ providing a `legacy-entries` attribute to the `jms-queue` and
...
-while the regular `entries` are for WildFly 10 JMS clients (using
+The `legacy-entries` must be used by legacy Jakarta Messaging clients (using HornetQ)
+while the regular `entries` are for WildFly 10 Jakarta Messaging clients (using
 Artemis).
 
-The legacy JMS client will then lookup this `legacy` JMS resources to
+The legacy Jakarta Messaging client will then lookup this `legacy` Jakarta Messaging resources to
 communicate with WildFly 10. +
-To avoid any code change in the legacy JMS clients, the legacy JNDI
-entries must match the lookup expected by the legacy JMS client.
```
[the remark this answers is not in the document]
Comment (ehsavoie) [reply docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_Backward_&_Forward_Compatibility.adoc:59]: or just use messaging

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_Backward_&_Forward_Compatibility.adoc [hunk 4]
```diff
@@ -71,25 +71,25 @@ in the legacy `messaging` subsystem and the regular entries will be
 created with a `-new` suffix. +
 If `add-legacy-entries` is set to `false` during migration, no legacy
 resources will be created in the `messaging-activemq` subsystem and
-legacy JMS clients will not be able to communicate with WildFly 10
+legacy Jakarta Messaging clients will not be able to communicate with WildFly 10
```
[the remark this answers is not in the document]
Comment (ehsavoie) [reply docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging_Backward_&_Forward_Compatibility.adoc:74]: or just use messaging

**Comment by RanabirChakraborty** [comment]
@ehsavoie Thanks! Did the changes accordingly :) please check once

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging.adoc [hunk 5]
```diff
@@ -397,30 +397,30 @@ WARNING: This feature **is primarily intended for development** as destinations
...
+== Jakarta Messaging Bridge
 
-The function of a JMS bridge is to consume messages from a source JMS
-destination, and send them to a target JMS destination. Typically either
+The function of a Jakarta Messaging bridge is to consume messages from a source Jakarta Messaging
+destination, and send them to a target Jakarta Messaging destination. Typically either
 the source or the target destinations are on different servers.
 The bridge can also be used to bridge messages from other non Artemis
-JMS servers, as long as they are JMS 1.1 compliant.
+Jakarta Messaging servers, as long as they are JMS 1.1 compliant.
```
[the remark this answers is not in the document]
Comment (bstansberry) [reply docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Messaging.adoc]: @ehsavoie Right now we should probably treat our docs as not being about WildFly Preview (i.e. where EE 9 variants of the specs come into the picture.) We'll probably need a major effort to deal with EE 9-specific changes like you've noted here.

**Pull request merged**